### PR TITLE
More work on katdal documentation

### PR DESCRIPTION
**Pull request by bmerry**
- Move the katdal module docstring into a "user guide" (@ludwigschwardt, not sure what you had in mind for duplicating this).
- Add the documentation of the v1 and v2 formats I found lying around.
- Add some brief stubs for v3 and v4 documentation.

### CODE REVIEW

File: doc/file_format.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+File format reference
```
Comment (ludwigschwardt) [inline doc/file_format.rst]: Data set format?

The system engineers might call this our Science Data Model (I got asked about this just two days ago 😄).

**Comment by ludwigschwardt** [comment]
On the module docs, it seems reasonably simple:
- In `__init__.py`, load `doc/intro.rst` and lob off the first two lines.
- Assign the string to `__doc__` and BOB's your uncle.

File: doc/mvf_v4.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
+MVF version 4 (MeerKAT)
+=======================
+
+The version 4 format is the standard format for MeerKAT visibility data.
+Unlike previous versions, the data for an observation does not reside in
+a single HDF5 file, as such files would be unmanageably large. Instead,
+the data is split into chunks, each in its own file, which are loaded
+from disk or the network on demand. For this reason, the term "dataset"
```
[the remark this answers is not in the document]
Comment (bmerry) [reply doc/mvf_v4.rst]: Except where you don't :-) But I'll change it.

File: doc/file_format.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+because katdal exists to hide these details and present a consistent,
+user-friendly view. This is reference documentation useful to katdal developers
+and to power users who need to extract information not presented by the katdal
+interface.
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Contents:
+
+   hdf5_format_v1
```
[the remark this answers is not in the document]
Comment (bmerry) [reply doc/file_format.rst]: That's just the name the file I found had. I've renamed them all.

File: doc/file_format.rst (hunk 1 above)
Comment (bmerry) [reply doc/file_format.rst]: Yep, changed to data set format.

File: doc/mvf_v4.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (ludwigschwardt) [reply doc/mvf_v4.rst]: Irritating, yes :frown:

[the remark this answers is not in the document]
Comment (mserylak) [reply doc/mvf_v4.rst]: One question. Would it be feasible to put small datasets (phase-ups, delay cals, short comm tests with sizes  < few tens of GBs) to HDF5 format? That would be great for portability i.e. when one would like to  work without access to VPN or offline etc.

[the remark this answers is not in the document]
Comment (bmerry) [reply doc/mvf_v4.rst]: We don't want to output to HDF5 again, but it's possible to use tools like rclone to sync all the data for an observation out of the archive into a local directory and load it from there. When I write more katdal docs I'll document the process.

**Comment by bmerry** [comment]
> In __init__.py, load doc/intro.rst and lob off the first two lines.

Where is it going to find `doc/intro.rst` from? It's not installed.

**Comment by ludwigschwardt** [comment]
Mmm, yes. More tinkering required.

**Pull request merged**